portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -46,10 +49,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    List<String> demoComments = new ArrayList<String>();
+    List demoComments = new ArrayList();
     for(Entity entity : results.asIterable()) {
-      String commentContent = (String) entity.getProperty(CONTENT_TEXT_PROPERTY_NAME);
-      demoComments.add(commentContent);
+      List<String> commentElementContent = new ArrayList<String>();
+
+      String commentText = (String) entity.getProperty(CONTENT_TEXT_PROPERTY_NAME);
+      String commentUserEmail = (String) entity.getProperty(USER_EMAIL_TEXT_PROPERTY_NAME);
+
+      commentElementContent.add(commentUserEmail);
+      commentElementContent.add(commentText);
+
+      demoComments.add(commentElementContent);
```

Comment:
Here is the example why we want to use protocol buffer in real industry. Both server and client needs to know the type, the size of the list and what each element in the list stands for, which is very difficult to maintain and debug. Without using the protocol buffer, you can alternatively put the data into a java class and serialize it into json string.

Again, you don't have to take my recommendation if the time is short, but try to use it in future development.

Reply:
The class solution is definitely more convenient like my AuthenticationInfo class, and I will look into using protocol buffers in the capstone project. Thank you!